core/shared/src/test/scala/cats/parse/ParserTest.scala

```diff
@@ -1086,6 +1108,66 @@ class ParserTest extends munit.ScalaCheckSuite {
     }
   }
 
+  property("rep parses at most max entries") {
+    val validMinMax = for {
+      min <- Gen.choose(1, Int.MaxValue)
+      max <- Gen.choose(min, Int.MaxValue)
+    } yield (min, max)
+    forAll(ParserGen.gen, validMinMax, Arbitrary.arbitrary[String]) { (genP, minmax, str) =>
+      {
+        val (min, max) = minmax
+        genP.fa.rep0(min, max).parse(str).foreach { case (_, l) =>
+          assert(l.length <= max)
+        }
+      }
+    }
+  }
+
+  property("rep0 parses at most max entries") {
+    val validMinMax = for {
+      min <- Gen.choose(1, Int.MaxValue)
+      max <- Gen.choose(min, Int.MaxValue)
+    } yield (min, max)
+    forAll(ParserGen.gen, validMinMax, Arbitrary.arbitrary[String]) { (genP, minmax, str) =>
+      {
+        val (min, max) = minmax
+        genP.fa.rep0(min, max).parse(str).foreach { case (_, l) =>
+          assert(l.length <= max)
+        }
+      }
+    }
+  }
+
+  object Counter extends Accumulator0[Unit, Int] {
```

Comment:
This is actually an interesting instance (maybe with Long) to make implicit.

Reply:
In test, or globally?

rep parsers can't successfully rep more than Int.MaxValue times on any input String, since String is limited to Int.MaxValue characters. Is making it Long to future-proof it for maybe parsing against something like fs2.Stream in the future worth it?

Reply:
I was thinking to put it in the companion to Accumulator0. Good point about Int vs Long.